File: craft/commands/rescue.py
import click

from craft.client import post
from craft.output import print_success


@click.command("enable")
@click.argument("vm_id")
def rescue_enable(vm_id):
    """Enable rescue mode (boots on next start)."""
    post(f"/vms/{vm_id}/rescue/enable")
    print_success("Rescue mode enabled. Start/reboot to enter rescue OS.")


@click.command("disable")
@click.argument("vm_id")
def rescue_disable(vm_id):
    """Disable rescue mode."""
    post(f"/vms/{vm_id}/rescue/disable")
    print_success("Rescue mode disabled. VM will boot normally.")

File: cmd/rescue.go
package cmd

import (
	"fmt"

	"github.com/CraftIntertech/craft-cli/internal/client"
	"github.com/CraftIntertech/craft-cli/internal/output"
	"github.com/spf13/cobra"
)

var rescueEnableCmd = &cobra.Command{
	Use:   "enable [VM_ID]",
	Short: "Enable rescue mode for a VM",
	Run: func(cmd *cobra.Command, args []string) {
		vmID := ""
		if len(args) > 0 {
			vmID = args[0]
		}
		if vmID == "" {
			vmID = SelectVM("Select VM to enable rescue mode")
		}

		result := client.Post(fmt.Sprintf("/vms/%s/rescue/enable", vmID), nil, true, 0)
		output.PrintSuccess("Rescue mode enabled.")
		output.PrintItem(extractObj(result))
	},
}

var rescueDisableCmd = &cobra.Command{
	Use:   "disable [VM_ID]",
	Short: "Disable rescue mode for a VM",
	Run: func(cmd *cobra.Command, args []string) {
		vmID := ""
		if len(args) > 0 {
			vmID = args[0]
		}
		if vmID == "" {
			vmID = SelectVM("Select VM to disable rescue mode")
		}

		result := client.Post(fmt.Sprintf("/vms/%s/rescue/disable", vmID), nil, true, 0)
		output.PrintSuccess("Rescue mode disabled.")
		output.PrintItem(extractObj(result))
	},
}

func init() {
	RescueCmd.AddCommand(rescueEnableCmd)
	RescueCmd.AddCommand(rescueDisableCmd)
}
